new/load-and-converge.md

```diff
@@ -0,0 +1,174 @@
+---
+RFC: unassigned
+Author: John Keiser <john@johnkeiser.com>
+Status: Draft
+Type: Standards Track
+---
+
+# Easy Resource Load And Converge
+
+With the introduction of `action` on resources, it becomes useful to have a
+blessed way to get the actual value of the resource. This proposal adds `load` and `converge` to help with this purpose, enabling:
+
+- Low-ceremony load methods (as easy to write as we can make it)
+- A super easy converge model that automatically compares current vs. desired
+  values and prints green text
+
+## Motivation
+
+    As a Chef resource writer,
+    I want to be able to read the current value of my resource at converge time,
+    so that it is easy to tell the difference between current and desired value.
+
+    As a Chef resource writer,
+    I want a converge model that compares current and desired values for me,
+    So that the easiest converge to write is the most correct one.
+
+## Specification
+
+### `load_actual_value`: in-place resource load
+
+When using `action`, one needs a way to load the *actual* system value of the resource, so that it can be compared to the desired value and a decision made as to whether to change anything.
+
+When the resource writer defines `load_actual_value` on the resource class, it can be called to load the real system value into the resource. Before any action runs, this will be used by `load_current_resource` to load the resource. `action` will do some important work before calling the new method:
+
+1. Create a new instance of the resource with the same name.
+2. Copy all non-desired-state values from the desired resource into the new instance.
+3. Call `load_actual_value` on the new instance.
+
+#### Non-existence
+
+To appropriately handle actual value loading, the user needs a way to specify that the actual value legitimately does not exist (rather than simply not filling in the object and getting `nil`s in it). If `load_actual_value` raises `Chef::Exceptions::ActualValueDoesNotExist`, the new resource will be discarded and `current_resource` becomes `nil`. The `actual_value_does_not_exist!` method can be called to raise this.
+
+NOTE: The alternative was to have users return `false` if the resource does not exist; but I didn't want users to be forced into the ceremony of a trailing `true` line.
+
+```ruby
+  load_actual_value do
+    # Check for existence before doing anything else.
+    actual_value_does_not_exist! if !File.exist?(path)
```

Comment:
This seems kind of unfortunate, but I agree I don't know of a better option. As an API it seems really silly though.